From patchwork Tue Nov  6 22:34:40 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Paolo Bonzini <pbonzini@redhat.com>
X-Patchwork-Id: 10671567
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id BB31313A4
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  6 Nov 2018 22:34:46 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id AE5AF2859F
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  6 Nov 2018 22:34:46 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id A27302ADF6; Tue,  6 Nov 2018 22:34:46 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.2 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI,SUBJ_OBFU_PUNCT_FEW autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 45A4E2859F
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  6 Nov 2018 22:34:46 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1730968AbeKGICK (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 7 Nov 2018 03:02:10 -0500
Received: from mx1.redhat.com ([209.132.183.28]:36840 "EHLO mx1.redhat.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1730835AbeKGICK (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 7 Nov 2018 03:02:10 -0500
Received: from smtp.corp.redhat.com (int-mx01.intmail.prod.int.phx2.redhat.com
 [10.5.11.11])
        (using TLSv1.2 with cipher AECDH-AES256-SHA (256/256 bits))
        (No client certificate requested)
        by mx1.redhat.com (Postfix) with ESMTPS id 5752D81254
        for <kvm@vger.kernel.org>; Tue,  6 Nov 2018 22:34:42 +0000 (UTC)
Received: from donizetti.redhat.com (ovpn-112-70.ams2.redhat.com
 [10.36.112.70])
        by smtp.corp.redhat.com (Postfix) with ESMTP id 9E51A600C4
        for <kvm@vger.kernel.org>; Tue,  6 Nov 2018 22:34:41 +0000 (UTC)
From: Paolo Bonzini <pbonzini@redhat.com>
To: kvm@vger.kernel.org
Subject: [PATCH kvm-unit-tests] pretty_print_stacks: port to Python 3
Date: Tue,  6 Nov 2018 23:34:40 +0100
Message-Id: <20181106223440.11869-1-pbonzini@redhat.com>
X-Scanned-By: MIMEDefang 2.79 on 10.5.11.11
X-Greylist: Sender IP whitelisted,
 not delayed by milter-greylist-4.5.16 (mx1.redhat.com [10.5.110.25]);
 Tue, 06 Nov 2018 22:34:42 +0000 (UTC)
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Recognize the line with a byte-based regex, then decode to Unicode
when printing the backtrace.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---
 scripts/pretty_print_stacks.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/pretty_print_stacks.py b/scripts/pretty_print_stacks.py
index 11042a7..a863c8e 100755
--- a/scripts/pretty_print_stacks.py
+++ b/scripts/pretty_print_stacks.py
@@ -37,14 +37,14 @@ def pretty_print_stack(binary, line):
         return
 
     for line in out.splitlines():
-        m = re.match('(.*) at [^ ]*/kvm-unit-tests/([^ ]*):([0-9]+)(.*)', line)
+        m = re.match(b'(.*) at [^ ]*/kvm-unit-tests/([^ ]*):([0-9]+)(.*)', line)
         if m is None:
             puts('%s\n' % line)
             return
 
         head, path, line, tail = m.groups()
         line = int(line)
-        puts('%s at %s:%d%s\n' % (head, path, line, tail))
+        puts('%s at %s:%d%s\n' % (head.decode(), path.decode(), line, tail.decode()))
         try:
             lines = open(path).readlines()
         except IOError:
